### PR TITLE
Cleanup in creation of secret for data transfers

### PR DESCRIPTION
**Pull request by ahadas**
follow-up on #195

### CODE REVIEW

File: pkg/controller/plan/kubevirt.go [hunk 1]
```diff
@@ -1644,7 +1650,7 @@ func vmOwnerReference(vm *cnv.VirtualMachine) (ref meta.OwnerReference) {
 
 // TODO move elsewhere
 func (r *KubeVirt) createOpenStackVolumes(vm ref.Ref) (err error) {
-	secret, err := r.ensureSecret(vm, false)
+	secret, err := r.ensureSecret(vm, r.secretDataSetterForCDI(vm))
```
Comment (liranr23) [inline pkg/controller/plan/kubevirt.go:1653]: CDI doesn't support OSP. but we have singular usage of the builder. that's why it was ok with false.

Comment (ahadas) [inline pkg/controller/plan/kubevirt.go:1653]: for openstack it doesn't make much sense as you clone the data and we don't use CDI.. I'll change that

Comment (ahadas) [reply pkg/controller/plan/kubevirt.go]: sure, it's not that it didn't work - but it was misleading, we actually do want to clone the provider secret for openstack as well, now that the code is more explicit we can see that

File: pkg/controller/plan/adapter/openstack/builder.go [hunk 2]
```diff
@@ -216,19 +216,6 @@ func (r *Builder) ResolvePersistentVolumeClaimIdentifier(pvc *core.PersistentVol
 	return ""
 }
 
-// Build credential secret.
-func (r *Builder) Secret(_ ref.Ref, in, secret *core.Secret) (err error) {
```
Comment (liranr23) [inline pkg/controller/plan/adapter/openstack/builder.go:220]: you need to have this function and just return for the builder API.

Comment (ahadas) [reply pkg/controller/plan/adapter/openstack/builder.go:220]: yeah, already done

**Review by liranr23 — APPROVED** [review]

**Review by bennyz — APPROVED** [review]

**Pull request merged**